python-modules/cis_change_service/cis_change_service/api.py

```diff
@@ -0,0 +1,83 @@
+import logging
+from flask import Flask
+from flask import jsonify
+from flask import request
+from flask_cors import CORS
+from flask_cors import cross_origin
+from cis_change_service import get_config
+from cis_change_service import profile
+from cis_change_service.idp import requires_auth
+from cis_change_service.idp import AuthError
+from cis_change_service import __version__
+
+from cis_publisher import operation
+
+
+app = Flask(__name__)
+config = get_config()
+logger = logging.getLogger(__name__)
+
+CORS(
+    app,
+    allow_headers=(
+        'x-requested-with',
+        'content-type',
+        'accept',
+        'origin',
+        'authorization',
+        'x-csrftoken',
+        'withcredentials',
+        'cache-control',
+        'cookie',
+        'session-id',
+    ),
+    supports_credentials=True
+)
+
+
+@app.errorhandler(AuthError)
+def handle_auth_error(ex):
+    response = jsonify(ex.error)
+    response.status_code = ex.status_code
+    return response
+
+
+@app.route('/')
+def index():
+    return 'Mozilla Change Integration Service Endpoint'
+
+
+@app.route('/version')
+def version():
+    response = __version__
+    return jsonify(message=response)
+
+
+@app.route('/change', methods=['GET', 'POST', 'PUT'])
+@cross_origin(headers=['Content-Type', 'Authorization'])
+@requires_auth
+def change():
+    user_profile = request.get_json(silent=True)
+    publish = operation.Publish()
+    result = publish.to_stream(user_profile)
+
+    if config('stream_bypass', namespace='cis', default='false') == 'true':
+        # Plan on stream integration not working an attempt a write directly to discoverable dynamo.
+        # Great for development, seeding the vault, and contingency.
+        logger.debug(
+            'Stream bypass activated.  Integrating user profile directly to dynamodb for: {}'.format(
+                user_profile.get('user_id').get('value')
```

Comment:
might want to use DotDicts here and elsewhere (Florian also uses them) so that you can write:
`user_profile.user_id.value`

if using `cis_profile` it supports this by default as well, which means you could do:

```
import cis_profile
...

def change():
    user_profile = cis_profile.User(request.get_json(silent=True))
    ....
    result = publisher.to_stream(user_profile.as_json())
    ...
    logger.debug(...user_profile.user_id.value)
```
etc

Reply:
cis_profile did did exist when I authored this.  Also ... I believe that user profile.validate() does raise as is and interrupts ... but this makes it easier to understand.

Reply:
yeah it does raise - i though about this a bit while writing it, and while i liked the idea of raising, it could also just return a valid or invalid result (only ldap2s3 uses it atm so it can still be changed, otherwise def validate_with_return ;-)

Reply:
Actually this _is_ the behavior.  But it's being handled in the operation class.  Catching the exception down there and raising a 400 back up to make sure we continue when processing a batch.  See L#61 in cis_publisher/operation.py

Reply:
ok :)